### DISCUSSION TITLE
Why is my Supabase RPC result: { documents: [], error: null } all the time???

### DISCUSSION THREAD

**Discussion by marius-manola** [supabase/supabase#26308] [Questions]
The first embeddingResponse call works fine, I get the embeddings, but why do I always get the error for the supabase rpc results? I tried setting the  match_threshold relatively low... I have info in my database. The tables are correctly assigned.

```
async function handleContext() {
    const supabase = createClient();
    const searchText = JSON.stringify(messages.slice(-1)[0]).replace(/\n/g, ' ');
  
    if (!searchText.trim()) return ""; 
  
    try {
      const headersList = headers();
      const origin = headersList.get('origin') || "";
  
      const embeddingResponse = await fetch(origin + "/api/chat/embeddings", {
        method: 'POST',
        body: JSON.stringify({ text: searchText }),
      });
  
      if (!embeddingResponse.ok) {
        throw new Error(`Embedding request failed with status: ${embeddingResponse.status}`);
      }
  
      const { embedding } = await embeddingResponse.json();
      console.log("embeddings:", embedding);
      
      const { data: documents, error } = await supabase.rpc("match_documents", {
        query_embedding: embedding,
        match_threshold: 0.5,
        match_count: 10,
      });
  
      console.log("Supabase RPC result:", { documents, error });
      
      if (error) throw error;
  
      return documents.map((doc: any) => doc.content).join("\n\n");
    } catch (error) {
      console.error("Error handling context:", error);
      return ""; // Return empty string in case of errors
    }
  }
```

here is the sql code for handling the match_documents function:
```
create or replace function match_documents (
  query_embedding vector(1536),
  match_threshold float,
  match_count int
)
returns table (
  id uuid,
  content text,
  similarity float
)
language sql stable
as $$
  select
    documents.id,
    documents.content,
    1 - (documents.embedding <=> query_embedding) as similarity
  from documents
  where 1 - (documents.embedding <=> query_embedding) > match_threshold
  order by similarity desc
  limit match_count;
$$;

```

**Comment by GaryAustin1** [comment]
Almost always rpc results of no error and empty array (especially if the function works in the SQL editor) means you don't meet RLS on the tables.  You could also try the function in impersonation mode in the SQL editor to test that.

**Reply by marius-manola** [reply]
I solved the issue when I turned down the RLS, because I put it to only allow for users to access the files that have the same id as their uuid.
When I disable the rls, the users' query gets matched with all the documents from the list, which I don't want
When I enable it, then the function doesn't work...

How could I allow only for users to perform the similarity search for documents that have the id as their uuid only?
Sorry if this is too basic.

**Reply by marius-manola** [reply]
all the files are stored within a list, where the columns have the id of the user who uploaded the text there

**Reply by marius-manola** [reply]
I also tried to change the sql function by adding:
  where documents.embedding <=> query_embedding < 1 - match_threshold
  and documents.id = auth.uid()

and still nothing..

**Reply by GaryAustin1** [reply]
If auth.uid() is not working then it is likely null, which means you don't really have a signed in user when you make the call.  That would also stop RLS from working on the id column matching auth.uid().

**Reply by marius-manola** [reply]
in both the profiles and documents tables, I can see the uuid assigned to the id column.
The user is logged in, since I am able to upload data from within my user profile to the list.
I have the user management starter query on.

Also, here's the RLS:

![Screenshot 2024-05-15 at 18 06 08](https://github.com/supabase/supabase/assets/168755267/67087f0f-9deb-4dc8-8a2a-e0a4a3f7d294)
